### PR TITLE
Add templates for .dsv files

### PR DESCRIPTION
**Pull request by jacobperron**
For more information about the introduction of .dsv files, see

- https://github.com/colcon/colcon-core/pull/209
- https://github.com/ament/ament_package/pull/89

I plan to use this change as part of the effort for getting ROS 2 Java working with ~~Dashing~~ and Eloquent.

### CODE REVIEW

**Comment by jacobperron** [comment]
@esteve PTAL. I believe this change is needed so the proper environment is set, e.g. when a colcon workspace is sourced.

**Comment by esteve** [comment]
@jacobperron do these supersede the `classpath.bat` / `classpath.sh` templates? If so, can they be removed in favor of .DSV?

**Comment by jacobperron** [comment]
Yes, these supersede the old .bat and .sh templates. But, I realized that this change is not compatible with Dashing (as the DSV format was introduced in Eloquent). I suggest we create dashing branch before this is merged.

**Comment by esteve** [comment]
@jacobperron done, I've created a `dashing` branch based off e6e23fbf54c2027999afa234474a4126175617cd

File: ament_java_resources/jni_library_path.dsv.template [hunk 1]
```diff
@@ -0,0 +1 @@
+prepend-non-duplicate;@JNI_LIB_ENV_VAR@;lib/jni
```
Comment (esteve) [inline ament_java_resources/jni_library_path.dsv.template:1]: @jacobperron just one question, `prepend-non-duplicate` is supposed work for both Linux and Windows, right?

Comment (jacobperron) [reply ament_java_resources/jni_library_path.dsv.template:1]: Yes, I'm pretty sure it's supposed to work for both.

**Comment by esteve** [comment]
@jacobperron thanks!

**Comment by esteve** [comment]
@jacobperron this is breaking master, so I had to revert it. The dashing branch is fairly big now, it'd make sense to merge it to master soon-ish and then target Eloquent/Foxy, otherwise it'll be more difficult to review the changes and keep track of all the work for both distros.

**Comment by jacobperron** [comment]
@esteve Is this still an issue now that we have a `dashing` branch for `ament_java`? I think we should have this PR on `master` as it's needed for Eloquent/Foxy (reapplying in https://github.com/ros2-java/ament_java/pull/15)

**Pull request merged**